### PR TITLE
Disable preemptive targeting for queued orders.

### LINKED ISSUE #16911: Plane queued orders don't work properly
They are on hold-fire stance

Yaks

Yaks2

### PR DESCRIPTION
**Pull request by tovl**
Fixes #16911

### CODE REVIEW

**Review by ghost — DISMISSED** [review]
Can't find any issues here.

**Comment by Punsho** [comment]
I dunno if it should be written as a separate issue but aircraft can attack only one target per flyby. It is a huge regression as you can no longer properly utilise shift-attack

**Comment by tovl** [comment]
Is this a regression in this PR or already on bleed?

**Comment by matjaeck** [comment]
>I dunno if it should be written as a separate issue but aircraft can attack only one target per flyby. It is a huge regression as you can no longer properly utilise shift-attack
>
> > Is this a regression in this PR or already on bleed?

Already present on bleed, but only visible under the right conditions. Do we want a new issue for this?

I have documented the behavior on latest bleed / PR / last release below:

<details>

<summary>292196e2d6683b23ac985797ce2ddab142753da5</summary>

#### 2 queued attack targets from idle state
- yak attacks the last queued target
- yak idles after destroying one target
queuefromidle


#### 2 queued attack targets from non-idle state
- yak attacks the first queued target
- **yak can not switch to next target in same flyby**
- yak returns to attack next target
queuedfromnonidle

</details>

<details>

<summary>This PR</summary>
The behavior in both cases is the same.

#### 2 queued attack targets from idle state
queuefromidle_pr

#### 2 queued attack targets from non-idle state
queuedfromnonidle_pr

- **yak can not switch to next target in same flyby**

</details>


<details>

<summary>Release-20190314</summary>
The behavior in both cases is the same.

#### 2 queued attack targets from idle state
queuefromidle_r


#### 2 queued attack targets from non-idle state
queuedfromnonidle_r

- **yak can switch to next target in same flyby**

- **yak can not switch to next target in same flyby**

</details>

File: OpenRA.Mods.Common/Traits/Attack/AttackBase.cs [hunk 1]
```diff
@@ -399,12 +399,11 @@ public void AttackTarget(Target target, bool queued, bool allowMove, bool forceA
...
 
-			if (!queued)
-				self.CancelActivity();
-
 			var activity = GetAttackActivity(self, target, allowMove, forceAttack, targetLineColor);
-			self.QueueActivity(activity);
-			OnQueueAttackActivity(self, activity, target, allowMove, forceAttack);
+			self.QueueActivity(queued, activity);
+
+			if (!queued)
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Attack/AttackBase.cs]: I expect we will need a similar check in `AttackBase.AttackTarget`.

**Comment by tovl** [comment]
Updated.

**Review by ghost — APPROVED** [review]
Fixes #16911, opened #16922 for the flyby issue.

**Review by pchote — APPROVED** [review]

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/d669926ab5a874f00aea4b2704a6293c6ee1b218

**Pull request merged**